auth/mtls/mtls.go

```diff
@@ -55,6 +58,70 @@ type CredentialsLoader interface {
 	// LoadServerCertificate returns the certificate that should be presented
 	// by the server to incoming clients.
 	LoadServerCertificate(context.Context) (tls.Certificate, error)
+
+	// CertRefreshed indicates if internally any of the cert data has
+	// been refreshed and should be reloaded. This will depend on the
+	// implementation to support but allows for dynamic refresh of certificates
+	// without a server restart.
+	CertsRefreshed() bool
+}
+
+type WrappedTransportCredentials struct {
+	creds      credentials.TransportCredentials
+	ctx        context.Context
+	loaderName string
+	loader     func(context.Context, string) (credentials.TransportCredentials, error)
+}
+
+func (w *WrappedTransportCredentials) checkRefresh() error {
+	loader, err := Loader(w.loaderName)
```

Comment:
do we actually need to call Loader() on each iteration? I suppose it's "correct", since something could technically register a new loader impl, but feels like we could just hang on to it after the first call

Reply:
It's more complicated. We don't know what kind of creds we're generating (client vs server) which is what loader is handling. The only interface we have for that has to lookup the loader each time then as we call it. I couldn't figure out another way here.

Reply:
ok, slept on it and more obvious. Fixed